src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -11155,7 +11155,40 @@ and TcNormalizedBinding declKind (cenv: cenv) env tpenv overallTy safeThisValOpt
 
         let supportEnforceAttributeTargets =
             (g.langVersion.SupportsFeature(LanguageFeature.EnforceAttributeTargets) && memberFlagsOpt.IsNone && not attrs.IsEmpty)
-            && not isVolatile // // VolatileFieldAttribute has a special treatment(specific error FS823)
+            && not isVolatile // VolatileFieldAttribute has a special treatment(specific error FS823)
+        
+        // Check for attributes in unit of measure expressions
+        // e.g. let x = 1.0<m>
+        //                  ^
+        let rec checkAttributeInMeasure ty =
```

Comment:
How is this block of code different from the one in `CheckDeclarations` ?
I understand we have to invoke the check separately for declarations of the types and then for usage for expression.

But the implementation of the check itself, does it differ internally?
I assume both are functions on `TType` which might lead to warnings/errors.

Reply:
These differ in that in `CehckExpressions.fs` we need to unwrap `TType` to get to `TType_app(typeInstantiation= [ TType_measure tm ])` while in the other we can work directly with `TType_measure`.